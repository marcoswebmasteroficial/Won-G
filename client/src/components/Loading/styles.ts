import styled, { css, keyframes } from 'styled-components'

const Animation = keyframes`
to {
  transform: rotate(1turn);
}
`
export const Wrapper = styled.div`
  width: 20em;
  height: 20em;
  font-size: 10px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
`
export const Face = styled.div`
  ${({ theme }) => css`
    position: absolute;
    border-radius: 50%;
    border-style: solid;
    animation: ${Animation} 3s linear infinite;

    &&:nth-child(1) {
      width: 100%;
      height: 100%;
      color: ${theme.colors.primary};
      border-color: currentColor transparent transparent currentColor;
      border-width: 0.2em 0.2em 0em 0em;
      --deg: -45deg;
      animation-direction: normal;
    }
    &&:nth-child(2) {
      width: 70%;
      height: 70%;
      color: ${theme.colors.secondary};
      border-color: currentColor currentColor transparent transparent;
      border-width: 0.2em 0em 0em 0.2em;
      --deg: -135deg;
      animation-direction: reverse;
    }
  `}
`
export const Text = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    opacity: 0.5;
  `}
`
export const Cicle = styled.div`
position: absolute;
width: 50%;
height: 0.1em;
top: 50%;
left: 50%;
background-color: transparent;
transform: rotate(var(--deg));
transform-origin: left;
&&::before {
  position: absolute;
  top: -0.5em;
  right: -0.5em;
  content: "";
  width: 1em;
  height: 1em;
  background-color: currentColor;
  border-radius: 50%;
  box-shadow: 0 0 2em, 0 0 4em, 0 0 6em, 0 0 8em, 0 0 10em,
    0 0 0 0.5em rgba(255, 255, 0, 0.1);
}
  }
`
