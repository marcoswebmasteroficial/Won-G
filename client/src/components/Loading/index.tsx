import * as S from './styles'

const Loading = () => (
  <S.Wrapper>
    <S.Face>
      <S.Cicle></S.Cicle>
    </S.Face>
    <S.Face>
      <S.Cicle></S.Cicle>
    </S.Face>
    <S.Text>Loading...</S.Text>
  </S.Wrapper>
)

export default Loading
