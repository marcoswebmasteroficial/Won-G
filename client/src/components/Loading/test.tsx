import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Loading from '.'

describe('<Loading />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Loading />)
    expect(screen.getByText(/Loading.../i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
