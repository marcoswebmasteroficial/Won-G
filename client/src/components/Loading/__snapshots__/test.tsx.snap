// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the heading 1`] = `
.c0 {
  width: 20em;
  height: 20em;
  font-size: 10px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  position: absolute;
  border-radius: 50%;
  border-style: solid;
  -webkit-animation: giFyqH 3s linear infinite;
  animation: giFyqH 3s linear infinite;
}

.c1.c1:nth-child(1) {
  width: 100%;
  height: 100%;
  color: #F231A5;
  border-color: currentColor transparent transparent currentColor;
  border-width: 0.2em 0.2em 0em 0em;
  --deg: -45deg;
  -webkit-animation-direction: normal;
  animation-direction: normal;
}

.c1.c1:nth-child(2) {
  width: 70%;
  height: 70%;
  color: #3CD3C1;
  border-color: currentColor currentColor transparent transparent;
  border-width: 0.2em 0em 0em 0.2em;
  --deg: -135deg;
  -webkit-animation-direction: reverse;
  animation-direction: reverse;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.6rem;
  opacity: 0.5;
}

.c2 {
  position: absolute;
  width: 50%;
  height: 0.1em;
  top: 50%;
  left: 50%;
  background-color: transparent;
  -webkit-transform: rotate(var(--deg));
  -ms-transform: rotate(var(--deg));
  transform: rotate(var(--deg));
  -webkit-transform-origin: left;
  -ms-transform-origin: left;
  transform-origin: left;
}

.c2.c2::before {
  position: absolute;
  top: -0.5em;
  right: -0.5em;
  content: "";
  width: 1em;
  height: 1em;
  background-color: currentColor;
  border-radius: 50%;
  box-shadow: 0 0 2em,0 0 4em,0 0 6em,0 0 8em,0 0 10em,0 0 0 0.5em rgba(255,255,0,0.1);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <p
    class="c3"
  >
    Loading...
  </p>
</div>
`;
